### PR TITLE
WIP: Attempt to replace a few nodes with myst nodes

### PR DESCRIPTION
**Pull request by aktech**
Currently it doesn't works, trying to progressive make changes via this Draft PR, not ready for review yet.
An attempt to tackle #190 

Notes:
  - Temporarily removed progress bar for debugging.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ test.dat
 test.npy
 test.out
 tree-sitter-rst/
+.idea/
```
Comment (Carreau) [inline .gitignore:15]: You want to put that in your global `.gitignore` https://gist.github.com/subfuzion/db7f57fff2fb6998a16c, not on each project you contribute to.

Comment (aktech) [reply .gitignore:15]: Ah, that's very useful!

File: papyri/crosslink.py [hunk 2]
```diff
@@ -392,7 +393,7 @@ def ingest(self, path: Path, check: bool) -> None:
             for k, v in doc_blob.content.items():
                 assert isinstance(v, Section), f"section {k} is not a Section: {v!r}"
             try:
-                doc_blob.validate()
+                papyri.common_ast.validate()
```
Comment (Carreau) [inline papyri/crosslink.py:396]: You'll need to pass an object here, or use a method.
```
papyri.common_ast.validate()
```

File: papyri/myst_ast.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+"""
+An attempt to create AST from MyST spec. At the moment just trying to
+add Text from Myst to the generated structure in the ts.py.
+
+"""
+from papyri.common_ast import Node
+
+
+# Prefixed with "M" as Text exists in take2.py as well
```
Comment (Carreau) [inline papyri/myst_ast.py:9]: 👍

File: papyri/crosslink.py (hunk 2 above)
Comment (Carreau) [reply papyri/crosslink.py:396]: I fixed it.

File: papyri/crosslink.py [hunk 4]
```diff
@@ -12,11 +12,11 @@
 import cbor2
 from there import print
 
+import papyri.common_ast
```
Comment (Carreau) [inline papyri/crosslink.py:15]: Try to get used to follow the style of the project you are in and not mix relative and pull import. Both are fine and have pros and cons, but you usually want to stick to the same style. 

Here it uses `from .XXXX` so you want to do `from . import common_ast`. 

Relative imports make it (a bit) easier to detect import cycles as you often to have `from ..` as you import a parent, and make it easier to not depend on your current package name.

**Comment by Carreau** [comment]
Couple of notes, not criticism, but maybe why you have issue moving forward:

You are biting more than you can crew (or even than I can chew) in a single PR and you get into a vicious circle because you are trying to do too many things.

Trying to put things in multiple module is a bit more complicated that doing all in a single module because of interdependencies, you end up with partially initialized module if you want to refer to myst_ast from take2 and vice versa. It is one of the (rare case), where you might have had premature optimisation. Not easy. 

It is ok when you know that some code is going to be removed to make it temporarily less correct, if you know it will be obviously and trivially correct in the end. 
That is to say it is ok to put reference to take2 nodes in myst_ast, as they will have to disappear once take2 (or specific instances) are deleted, though it needs a bit more structure in how you move forward.

If you follow the above, and you have a relatively clear plan in how to move forward, then it's easier to make a smaller step forward that work and pass CI and linter. Then you can iterate. 

It is finally ok to undo some work. Say you introduce MParagraph and you realize that it needs MStrong, it's ok to throw away the introduction of MParagrpah and restart on MStrong is MStrong is easier and you can make MStrong work on it's own. Then the Journey from MStrong that works to MStong+MParagraph is shorter and easier.

You will often see this summarized as "Make it work, Make it right, Make it fast", here we can forget about the "Make it fast" as fast is not our primary goal, but in general it's easier to have a working things first, then polish the edge case. In another way, we are the turtle here, not the hare :-)

Good work otherwise, i've deactivate numpy and scipy, but papyri shoudl be able to gen it's own doc, I'll see if I can make it ingest next, and then we can merge and iterate one small step at a time.

**Comment by Carreau** [comment]
Also side note, when I say a small step, I also mean the a small step can be a PR in itself. Should should feel ok make a PR that is even 5 minute of work if it's a step stone to something bigger.

**Comment by Carreau** [comment]
Self gen is passing on CI ! I'll fic ingest tomorrow.

File: papyri/crosslink.py (hunk 4 above)
Comment (aktech) [reply papyri/crosslink.py]: Totally agree, I had not done any cleaning on this PR, as of yet. These imports are auto generated by my IDE's one click import functionality.

Comment (aktech) [reply papyri/crosslink.py]: I usually do cleaning, when the general architecture/logic is approved.

**Comment by aktech** [comment]
Tests passing now!

**Pull request merged**